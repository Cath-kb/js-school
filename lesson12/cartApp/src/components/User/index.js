import template from './template.ejs';

export default ({name, email, phone, address, city, country}) => template({
  user: {name, email, phone, address, city, country},
});
