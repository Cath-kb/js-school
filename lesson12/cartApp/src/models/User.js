import api from './api';

const DEFAULT_USER = 1;

export async function getUser(userId = DEFAULT_USER) {
    try {
        const { data } = await api.get(`/users/${userId}`);

        return data;
    } catch (e) {
        throw e; // todo: handle error
    }
}