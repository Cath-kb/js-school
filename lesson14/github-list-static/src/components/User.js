import React from 'react';

class User extends React.Component {
  render() {
    const {avatar, person} = this.props.user;

    return (
      <div className="row my-4">
        <div className="col-sm-2" data-id="user">
          <img src={avatar.url} alt="User Avatar" className="w-100 img-fluid rounded mb-4 mb-sm-0" />
        </div>
        <div className="col-sm-10">
          <h2 className="d-flex align-items-center">
            {person.login}
            {person.isAdmin && <span className="ml-4 badge badge-pill badge-dark">admin</span>}
          </h2>
          <h3><a href={person.url} target="_blank" title="Go to user profile">{person.url}</a></h3>
        </div>
      </div>
    );
  }
}

export default User;
