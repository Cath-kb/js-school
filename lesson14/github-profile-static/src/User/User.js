import React from "react";
import Avatar from "./Avatar";
import Person from "./Person";
import Statistics from "./Statistics";

class User extends React.Component {
  render() {
    const {avatar, person, statistics} = this.props.user;
    return (
      <div className="row">
        <div className="col-md-6  mb-4">
          <Avatar avatar={avatar} />
        </div>
        <div className="col-md-6">
          <Person person={person} />
          <Statistics data={statistics} />
        </div>
      </div>
    );
  }
}

export default User;
