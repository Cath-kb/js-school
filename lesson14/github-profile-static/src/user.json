{
  "login": "Cath-kb",
  "id": 2164575,
  "node_id": "MDQ6VXNlcjIxNjQ1NzU=",
  "avatar_url": "https://avatars3.githubusercontent.com/u/2164575?v=4",
  "gravatar_id": "",
  "url": "https://api.github.com/users/Cath-kb",
  "html_url": "https://github.com/Cath-kb",
  "followers_url": "https://api.github.com/users/Cath-kb/followers",
  "following_url": "https://api.github.com/users/Cath-kb/following{/other_user}",
  "gists_url": "https://api.github.com/users/Cath-kb/gists{/gist_id}",
  "starred_url": "https://api.github.com/users/Cath-kb/starred{/owner}{/repo}",
  "subscriptions_url": "https://api.github.com/users/Cath-kb/subscriptions",
  "organizations_url": "https://api.github.com/users/Cath-kb/orgs",
  "repos_url": "https://api.github.com/users/Cath-kb/repos",
  "events_url": "https://api.github.com/users/Cath-kb/events{/privacy}",
  "received_events_url": "https://api.github.com/users/Cath-kb/received_events",
  "type": "User",
  "site_admin": false,
  "name": "Catherine Chernysheva",
  "company": null,
  "blog": "https://www.linkedin.com/in/cathkb",
  "location": "Kyiv, Ukraine",
  "email": null,
  "hireable": null,
  "bio": null,
  "public_repos": 21,
  "public_gists": 1,
  "followers": 4,
  "following": 0,
  "created_at": "2012-08-16T17:44:34Z",
  "updated_at": "2018-08-05T08:14:18Z"
}
