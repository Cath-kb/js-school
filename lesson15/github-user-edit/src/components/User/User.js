import React from 'react';

import Avatar from "./Avatar";
import Person from "./Person";
import Statistics from "./Statistics";

class User extends React.Component {
  render() {
    const { onClick } = this.props;
    const {avatar, person, statistics } = this.props.user;

    return (
      <div className="row">
        <div className="col-md-4 text-center text-md-left mb-4">
          <Avatar avatar={avatar} />
        </div>
        <div className="col-md-8">
          <Person person={person} onClick={onClick} />
          <Statistics data={statistics} />
        </div>
      </div>
    );
  }
}

export default User;
