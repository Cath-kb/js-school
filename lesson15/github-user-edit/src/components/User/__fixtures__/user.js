const user = {
  avatar: {
    url: 'https://avatars3.githubusercontent.com/u/2164575?v=4',
  },
  person: {
    name: 'John Doe',
    login: 'joe',
    company: '',
    location: 'Aldebaran',
  },
  statistics: {
    repos: 5,
    following: 0,
    followers: 5,
  }
};

export default user;