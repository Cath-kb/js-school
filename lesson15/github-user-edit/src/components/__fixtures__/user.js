const user = {
  avatar: {
    url: 'https://avatars3.githubusercontent.com/u/2164575?v=4',
  },
  person: {
    name: 'John Doe',
    login: 'joe',
    company: '',
    location: 'Aldebaran',
    origin: 'https://github.com/joe',
  },
  statistics: {
    repos: 5,
    following: 0,
    followers: 5,
  }
};

export default user;