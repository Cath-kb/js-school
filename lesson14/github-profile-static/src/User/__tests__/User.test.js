import React from 'react';
import ReactDOM from 'react-dom';
import User from './../User';
import user from './../__fixtures__/user';
import renderer from "react-test-renderer";

describe('User', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<User user={user}>
    </User>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should be rendered without errors', () => {
    const html = renderer.create(<User user={user}/>).toJSON();
    expect(html).toMatchSnapshot();
  });
});
