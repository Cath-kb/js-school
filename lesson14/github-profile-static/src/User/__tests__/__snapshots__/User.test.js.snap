// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User should be rendered without errors 1`] = `
<div
  className="row"
>
  <div
    className="col-md-6  mb-4"
  >
    <img
      alt="User Avatar"
      className="img-fluid mx-auto d-block rounded"
      src="https://avatars3.githubusercontent.com/u/2164575?v=4"
    />
  </div>
  <div
    className="col-md-6"
  >
    <div>
      <h2>
        John Doe
      </h2>
      <h3>
        joe
      </h3>
      
      <p>
        Aldebaran
      </p>
    </div>
    <div>
      <h4>
        <strong>
          Statistics:
        </strong>
      </h4>
      <ul
        className="list-group"
      >
        <li
          className="list-group-item d-flex justify-content-between align-items-center"
        >
          Repositories
          : 
          <span
            className="badge badge-primary badge-pill"
          >
            5
          </span>
        </li>
        <li
          className="list-group-item d-flex justify-content-between align-items-center"
        >
          Following
          : 
          <span
            className="badge badge-primary badge-pill"
          >
            0
          </span>
        </li>
        <li
          className="list-group-item d-flex justify-content-between align-items-center"
        >
          Followers
          : 
          <span
            className="badge badge-primary badge-pill"
          >
            5
          </span>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
